### ISSUE TITLE
Detect code changes based on filesystem events rather than polling

### ISSUE THREAD

**Issue by tbekolay** [nengo/nengo_viz#182]
There's a Python package for listening to filesystem change events called [watchdog](https://pypi.python.org/pypi/watchdog). It uses platform specific and platform-independent ways of detecting when files change, which should be more efficient than the polling approach used in #161. It's another dependency, but it shouldn't be hard to install; alternatively, we can vendorize it, or just take the bits we need.

**Comment by hunse** [comment]
Looks cool! Some of the core code appears to be in C, though, so I'm not sure if that makes it more difficult to install. (Does `pip` precompile for different platforms, or always compile individually?) Says it requires XCode on OS X.

**Comment by tcstewar** [comment]
I could definitely see something like this if it didn't require a big dependency...  The C dependency really scares me, though, especially with things like trying to get it to run in embedded systems.  So for now I think the extra cost of pinging the file system every half second isn't too bad to pay.

**Comment by jgosmann** [comment]
Another option would be to use such a library, but have a fall back to polling if it cannot be installed or fails for other reasons. But then the question is whether it is worth the added code complexity.

**Comment by tcstewar** [comment]
> Another option would be to use such a library, but have a fall back to polling if it cannot be installed or fails for other reasons. But then the question is whether it is worth the added code complexity.

Good point!  I hadn't thought of just doing it as a fall-back....  but yeah, the added code complexity seems a bit weird. 

This is probably a situation where we should actually do some profiling to see if it's even worth worrying about....   :)
